tests/spec/urlparser-spec.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Test for URL Parameter parsing
+ */
+
+
+let defaultUrl = "https://python.microbit.org/v/1.1"; 
+describe("Testing different URL formats", () => {
+    test("Check if no Parameter is recognised", () =>{
+        expect(global.urlparse("parameter", defaultUrl) === undefined).toBeTruthy();
+    });
+    test("Check if URL fragments are ignored", ()=>{
+        expect(global.urlparse("anchor", defaultUrl + "#anchor") === undefined).toBeTruthy();
+    });
+    test("Check if parameter is recognised", () =>{
+        expect(global.urlparse("key", defaultUrl + "?key=value1") === "value1").toBeTruthy();
+    });
+    test("Check if more than one parameter affects result", () =>{
+        expect(global.urlparse("key1", defaultUrl + "?key1=value1&key2=value2") === "value1").toBeTruthy();
+    });
+    test("Check if more than one url parameter affects result and that URL fragments are ignored", ()=>{
+        expect(global.urlparse("key2", defaultUrl + "?key1=value1&key2=value2#anchor") === "value2").toBeTruthy();
+    });
+    test("Check if incomplete url fragments (just contain keys) and a URL fragments work", ()=>{
+        expect(global.urlparse("key2", defaultUrl + "?key1&key2#anchor") === null).toBeTruthy();
+    });
+    test("Check if incomplete URL parameter returns null", ()=>{
+        expect(global.urlparse("anchor", defaultUrl + "?key1=") === null).toBeTruthy();
```

Comment:
In this case you are looking for a key named `anchor`, which is not in the URL. Perhaps you meant to get the key `key1` instead?

Reply:
Apart from that, it would be good to have a test for trying to get a key not present in the URL when there are other keys present (as `getUrlParam()` has a different code path for this test case if keys are or not present):

```js
expect(global.urlparse("notpresent", defaultUrl + "?key1=value").toBeUndefined();
```

Reply:
This was a typo error, it's fixed now